Fix: Running the "make cert-manager-self-signed-issuer-create" in step 7 fails

avisiedo opened the pull request:
- Add removed rules to mk/cert-manager.mk to create/delete self-signed certificate issuer.
- Update samples-* rules to use a single file.
- Add 'private.mk.example' to quickly create 'private.mk' file.
- Update 'Quickstart' and 'Deploying with OLM' sections.

Fix issue: https://github.com/freeipa/freeipa-operator/issues/54

avisiedo committed:
Add 'private.mk.example' to quickly create 'private.mk' file

avisiedo committed:
Add removed rules to mk/cert-manager.mk to create/delete self-signed certificate issuer

avisiedo committed:
Update samples-* rules to use a single file

avisiedo committed:
Update 'Quickstart' and 'Deploying with OLM' sections

avisiedo committed:
Fix linter hint on 'README.md' file

vashirov commented on mk/cert-manager.mk:
```diff
@@ -10,3 +10,10 @@ cert-manager-install:  ## Install cert-manager operator
 
+.PHONY: cert-manager-self-signed-issuer-create
+cert-manager-self-signed-issuer-create:  ## Create a cluster self signed issuer
+	kubectl create -f config/certmanager/clusterissuer-selfsigned.yaml
```
Since we're targeting OpenShift platform and not a vanilla kubernetes distribution (we depend on routes, SCCs and maybe some other OpenShift-specific resources), I think we should stick with `oc` and use it everywhere.

avisiedo replied:
Updated! thanks!

avisiedo committed:
review: A few changes observed during pr review

- Update README.md steps indicating to be logged in
  as cluster-admin user, and reference to install
  the CRD into the cluster before run the controller
  locally.
- Launch operator-sdk binary from ./bin directory.
- Update some occurrencies of kubectl by oc.
- Bumped cert-manager to 1.7.2 version.

vashirov commented on README.md:
```diff
@@ -31,10 +31,20 @@ Experimental freeipa-operator for Freeipa.
-1. Now create a new namespace by: `kubectl create namespace my-freeipa`
+1. Now create a new namespace by: `oc create namespace my-freeipa`
+
+1. As clust-admin user logged in the cluster run:
```
nit: cluster-admin

avisiedo replied:
thanks :) updated!

avisiedo committed:
typo: cluster-admin at README.md

avisiedo committed:
doc: Update and add steps at Deploying with OLM

- Use 'oc delete project ..'
- Create scc profile in early steps.
- Delete scc profile in later steps.

vashirov commented:
/lgtm

vashirov commented:
/lgtm

vashirov commented:
/approve
